analysis/deep2_r23_o32/zoom_and_gauss_general.py

```diff
@@ -198,235 +214,203 @@ def zoom_gauss_plot(dataset, fitspath, tab, stack2D, dispersion, s2, wave,
+            t_ax.set_xlim(x1 + 50, x2 - 50)
 
             if line_name == 'OIII_4363':
-                t_ax.set_ylim(0,1)
+                t_ax.set_ylim(0, 1)
             
-            if dataset == 'Grid' or dataset=='O32_Grid' or dataset =='R23_Grid' \
-                    or dataset =='Double_Bin' or dataset =='n_Bins':
+            if dataset == 'Grid' or dataset == 'O32_Grid' or dataset == 'R23_Grid' \
+                    or dataset == 'Double_Bin' or dataset == 'n_Bins':
                 if line_type == 'Balmer': 
-                    txt0  = 'Line: %.3f, ID: %i, R_23: %.3f O_32: %.3f\n' % (o1[0], ID[rr], asc_tab['logR23_min'][rr], asc_tab['logO32_min'][rr]) + '\n'
-                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %.3f\n' % (RMS_tot, RMS_pix, N_gal_array[rr]) 
-                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f'% (o1[3], o1[1], max0) + '\n'
-                    txt0 += 'o1[2]: %.3f o1[4]: %.3f  o1[5]: %.3f'% (o1[2], o1[4], o1[5]) + '\n'
-                    txt0 += 'F_G: %.3f F_S: %.3f' %(flux_g, flux_s) + '\n'
-                    txt0 += 'S/N: %.3f' %(SN_array[rr])
-
-                if line_type == 'Single' or line_type =='Oxy2': 
-                    txt0  = 'Line: %.3f, ID: %i, R_23: %.3f O_32: %.3f\n' % (o1[0], ID[rr], asc_tab['logR23_min'][rr], asc_tab['logO32_min'][rr]) + '\n'
-                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %i\n' % (RMS_tot, RMS_pix, N_gal_array[rr]) 
-                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f o1[2]: %.3f'% (o1[3], o1[1], max0, o1[2]) + '\n'
-                    txt0 += 'F_G: %.3f F_S: %.3f' %(flux_g, flux_s) + '\n'
-                    txt0 += 'S/N: %.3f' %(SN_array[rr])
+                    txt0 = 'Line: %.3f, ID: %i, R_23: %.3f O_32: %.3f\n' % \
+                            (o1[0], id[rr], asc_tab['logR23_min'][rr], asc_tab['logO32_min'][rr]) + '\n'
+                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %.3f\n' % (rms_tot, rms_pix, N_gal_array[rr])
+                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f' % (o1[3], o1[1], max0) + '\n'
+                    txt0 += 'o1[2]: %.3f o1[4]: %.3f  o1[5]: %.3f' % (o1[2], o1[4], o1[5]) + '\n'
+                    txt0 += 'F_G: %.3f F_S: %.3f' % (flux_g, flux_s) + '\n'
+                    txt0 += 'S/N: %.3f' % (SN_array[rr])
+
+                if line_type == 'Single' or line_type == 'Oxy2':
+                    txt0 = 'Line: %.3f, ID: %i, R_23: %.3f O_32: %.3f\n' % \
+                            (o1[0], id[rr], asc_tab['logR23_min'][rr], asc_tab['logO32_min'][rr]) + '\n'
+                    txt0 += 'RMS: %.3f RMS/pix: %.3f, N: %i\n' % (rms_tot, rms_pix, N_gal_array[rr])
+                    txt0 += r'Median: %.3f $\sigma$: %.3f  Norm: %.3f o1[2]: %.3f' % (o1[3], o1[1], max0, o1[2]) + '\n'
+                    txt0 += 'F_G: %.3f F_S: %.3f' % (flux_g, flux_s) + '\n'
+                    txt0 += 'S/N: %.3f' % (SN_array[rr])
```

Comment:
You might be able to simplify given repetition and only have if statements for differences

Reply:
I know, but this is easier for me to edit if we need to change the text on the graph. I will try to think of a different way and implement it later.